src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -43,35 +42,53 @@ def initialise(self):
 
     def step(self, t, t_new, current_rv, **kwargs):
         """Gaussian IVP filter step as nonlinear Kalman filtering with zero data."""
+        # 0. Obtain the diffusion matrix; required for calibration / error estimation
+        discrete_dynamics = self.gfilt.dynamod.discretise(t_new - t)
+        diffmat = discrete_dynamics.diffusionmatrix(t_new)
+
+        # 1. Predict
         pred_rv, _ = self.gfilt.predict(t, t_new, current_rv, **kwargs)
+
+        # 2. Measure
+        meas_rv, info = self.gfilt.measmod.transition_rv(pred_rv, t_new, **kwargs)
+
+        # 3. Estimate the diffusion (sigma squared)
+        self.sigma_squared_mle = self._estimate_diffusion(pred_rv, meas_rv)
+        # 3.1. Adjust the prediction covariance to include the diffusion
+        pred_rv = Normal(
+            pred_rv.mean, pred_rv.cov + (self.sigma_squared_mle - 1) * diffmat
+        )
+        # 3.2 Update the measurement covariance (measure again)
+        meas_rv, info = self.gfilt.measmod.transition_rv(pred_rv, t_new, **kwargs)
+
+        # 4. Update
         zero_data = 0.0
         filt_rv, meas_rv, info = self.gfilt.update(t_new, pred_rv, zero_data, **kwargs)
-        errorest, self.sigma_squared_current = self._estimate_error(
-            filt_rv, meas_rv, info
-        )
-        return filt_rv, errorest
 
-    def method_callback(self, time, current_guess, current_error):
-        """Update the sigma-squared (ssq) estimate."""
-        self.sigma_squared_global = (
-            self.sigma_squared_global
-            + (self.sigma_squared_current - self.sigma_squared_global) / self.num_steps
+        # 5. Error estimate
+        local_errors = self._estimate_local_error(
+            pred_rv, t_new, self.sigma_squared_mle * diffmat
         )
+        err = np.linalg.norm(local_errors)
+
+        return filt_rv, err
 
     def postprocess(self, times, rvs):
         """
         Rescale covariances with sigma square estimate,
         (if specified) smooth the estimate, return ODESolution.
         """
-        rvs = self._rescale(rvs)
+        if False:  # pylint: disable=using-constant-test
+            # will become useful again for time-fixed diffusion models
```

Comment:
thanks for this reminder, I think it will be useful in the future! 

Though (and this is by no means criticism of this occasion here!) it seems to me that those kind of hotfixes get more and more common, no? Maybe we keep an eye on this...

Reply:
I can happily delete the whole `if` branch and add it later again. I just thought it's cleaner to keep it there since the ultimate goal is to implement a fixed diffusion model very soon, which requires that `if` but with some _real_ condition. Should I just remove it for now?

Reply:
no, as I said: I think it will be useful in the future :) That was just a side-remark